### PR TITLE
Extract publication logic to buildSrc

### PR DESCRIPTION
**Pull request by blindpirate**
After introducing `client-protocol` subproject, we need to publish
more than 1 project, so we extract the common logic to buildSrc.

### CODE REVIEW

File: buildSrc/src/main/kotlin/profiler.publication.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+publishing {
+    publications {
+        register<MavenPublication>("mavenJava") {
+            from(components["java"])
+
+            if (project.parent == null) {
+                afterEvaluate {
+                    val profilerDistribution = artifacts.add("archives", tasks.named<Zip>("distZip").flatMap { it.archiveFile }) {
```
Comment (wolfs) [inline buildSrc/src/main/kotlin/profiler.publication.gradle.kts:14]: This should go into the root project. I don't see why the general publication plugin would add the zip.

**Review by wolfs — CHANGES_REQUESTED** [review]

File: build.gradle.kts [hunk 2]
```diff
@@ -141,33 +141,13 @@ val profilerDistribution = artifacts.add("archives", tasks.distZip.flatMap { it.
...
-            credentials {
-                username = project.findProperty("artifactoryUsername") as String?
-                password = project.findProperty("artifactoryPassword") as String?
-            }
         }
     }
 }
 
-fun Project.gradleInternalRepositoryUrl(): URI {
+fun Project.gradleInternalRepositoryUrl(): java.net.URI {
```
Comment (wolfs) [inline build.gradle.kts:150]: Any reason why we don't use the import here any more?

**Review by wolfs — APPROVED** [review]
LGTM! I left one comment.

File: build.gradle.kts (hunk 2 above)
Comment (blindpirate) [reply build.gradle.kts:150]: This is an IDEA auto-format issue, will fix.

**Pull request merged**